ui/src/pages/version/HistoryDetails.js

```diff
@@ -0,0 +1,182 @@
+import { DateFromNow } from "@caraml-dev/ui-lib";
+import {
+  EuiBadge,
+  EuiButtonIcon,
+  EuiCodeBlock,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiHealth,
+  EuiInMemoryTable,
+  EuiScreenReaderOnly,
+  EuiText,
+} from "@elastic/eui";
+import { useState } from "react";
+import { ConfigSection, ConfigSectionPanel } from "../../components/section";
+import { useMerlinApi } from "../../hooks/useMerlinApi";
+
+const defaultTextSize = "s";
+
+const DeploymentStatus = ({
+  status,
+  deployment,
+  deployedRevision,
+  endpointStatus,
+}) => {
+  if (deployment.error !== "") {
+    return <EuiHealth color="danger">Failed</EuiHealth>;
+  }
+
+  if (status === "running" || status === "serving") {
+    if (
+      deployment.id === deployedRevision.id &&
+      (endpointStatus === "running" || endpointStatus === "serving")
+    ) {
+      return <EuiHealth color="success">Deployed</EuiHealth>;
+    }
+    return <EuiHealth color="default">Not Deployed</EuiHealth>;
+  } else if (status === "pending") {
+    return <EuiHealth color="gray">Pending</EuiHealth>;
+  }
+};
+
+const RevisionPanel = ({ deployments, deploymentsLoaded, endpoint }) => {
+  const orderedDeployments = deployments.sort((a, b) => b.id - a.id);
+
+  const deployedRevision = orderedDeployments.find(
+    (deployment) =>
+      deployment.status === "running" || deployment.status === "serving"
+  ) || { id: null };
+
+  const canBeExpanded = (deployment) => {
+    return deployment.error !== "";
+  };
+
+  const [itemIdToExpandedRowMap, setItemIdToExpandedRowMap] = useState({});
+
+  const toggleDetails = (deployment) => {
+    const itemIdToExpandedRowMapValues = { ...itemIdToExpandedRowMap };
+
+    if (itemIdToExpandedRowMapValues[deployment.id]) {
+      delete itemIdToExpandedRowMapValues[deployment.id];
+    } else {
+      itemIdToExpandedRowMapValues[deployment.id] = (
+        <>
+          <EuiText className="expandedRow-title" size="xs">
+            Error message
+          </EuiText>
+          <EuiCodeBlock isCopyable>{deployment.error}</EuiCodeBlock>
+        </>
+      );
+    }
+    setItemIdToExpandedRowMap(itemIdToExpandedRowMapValues);
+  };
+
+  const cellProps = (item, column) => {
```

Comment:
Haha this is really nice and helpful; maybe we can bring this into some utils package in the future and reuse it in other places 🙏